inst/rmarkdown/templates/cholera_outbreak/skeleton/skeleton.Rmd

```diff
@@ -568,6 +638,27 @@ linelist_cleaned <- linelist_cleaned %>%
 #     "Village D" = "Villiage D"
 #   ))
 
+## sometimes, coding is inconsistent across variables -- for example, "Yes" / "No"
+## may be coded as Y, y, yes, 1 / N, n, no, 0. You can change them all at once!
+## Create a list of the variables you want to change, and run the following.
+## You may need to edit this code if options are different in your data.
+
+# # create list of variables
+# change_yn <- c("headache", "fever", "vomiting")
+# 
+# # standardize options
+# linelist_cleaned <- linelist_cleaned %>%
+#   mutate_at(vars(change_yn), ~case_when(
+#     . == "Yes" ~ "Yes",
+#     . == "y"   ~ "Yes",
+#     . == "Y"   ~ "Yes",
+#     . == "No"  ~ "No",
+#     . == "N"   ~ "No",
+#     . == "n"   ~ "No",
+#     . == 1     ~ "Yes",
+#     . == 0     ~ "No",
+#     TRUE       ~ "Unknown"
+#     ))
```

Comment:
Sooooooooo, I'm a bit of a dumdum. You don't need this convoluted `case_when()` stuff here. We can simply use forcats:

```r
  mutate_at(vars(change_yn), forcats::fct_collapse, 
    Yes = c("y", "Y", "Yes", "1", "yes"),
    No  = c("n", "N", "No", "0", "no"),
    group_other = TRUE
  ) %>%
  mutate_at(vars(change_yn), forcats::fct_recode, Unknown = "Other")
```

Reply:
Doesn't work for me:
```
linelist_cleaned1 <- linelist_cleaned %>%
    mutate_at(vars(change_yn), forcats::fct_collapse, 
    Yes = c("y", "Y", "Yes", "1", "yes"),
    No  = c("n", "N", "No", "0", "no"),
    group_other = TRUE
  ) %>%
  mutate_at(vars(change_yn), forcats::fct_recode, Unknown = "Other")

Error in names(levs) <- names(new)[rep(seq_along(new), vapply(new, length,  : 
  'names' attribute [12] must be the same length as the vector [5]
```

Reply:
oh how fun -_-

Okay in that case, we might be able to get away with something like this:

```r
 linelist_cleaned1 <- linelist_cleaned %>%  
      mutate_at(vars(change_yn), forcats::fct_recode,   
      Yes = "y",  
      Yes = "Y",  
      Yes = "yes",  
      Yes = "1",   
      No  = "n",  
      No  = "N",  
      No  = "no",  
      No  = "0",  
    )     
```

This is less tedious than the `case_when` and more consistent with the NEW = OLD syntax.

Reply:
I'm getting an error, but I think it's generally okay:
```
test <- c("1", "0", "y", "Yes", "NO")
linelist_cleaned$test <- sample(test, replace = TRUE, size = 300)
linelist_cleaned$test2 <- sample(test, replace = TRUE, size = 300)

change_yn <- c("test", "test2")

 linelist_cleaned1 <- linelist_cleaned %>%  
      mutate_at(vars(change_yn), forcats::fct_recode,   
      Yes = "y",  
      Yes = "Y",  
      Yes = "yes",  
      Yes = "1",   
      No  = "n",  
      No  = "N",  
      No  = "no",  
      No  = "0",  
    ) 

Unknown levels in `f`: Y, yes, n, N, noUnknown levels in `f`: Y, yes, n, N, no
```
So that's just telling me that there are no values in test or test2 cols that are Y, yes, n, N, or no -- which makes sense as I didn't code them in. And it's also not grabbing my "NO" values since it's not in the list. Does that sound right?

Reply:
Those aren't errors, those are warnings, so it's okay.

Reply:
The idea is that if the user finds that something isn't being grabbed, they can include it in the list themselves, so they would add No = "NO", there as well